src/main/java/hudson/remoting/AbstractByteBufferCommandTransport.java

```diff
@@ -293,14 +326,23 @@ public final void write(Command cmd, boolean last) throws IOException {
             int frame = remaining > transportFrameSize
                     ? transportFrameSize
                     : (int) remaining; // # of bytes we send in this chunk
-            ((Buffer) writeChunkHeader).clear();
-            ChunkHeader.write(writeChunkHeader, frame, remaining > transportFrameSize);
-            ((Buffer) writeChunkHeader).flip();
-            ((Buffer) writeChunkBody).clear();
-            ((Buffer) writeChunkBody).limit(frame);
-            sendStaging.get(writeChunkBody);
-            ((Buffer) writeChunkBody).flip();
-            write(writeChunkHeader, writeChunkBody);
+            if (writeChunkHeader == null) {
+                ((Buffer) writeChunkCombined).clear();
+                ChunkHeader.write(writeChunkCombined, frame, remaining > transportFrameSize);
+                ((Buffer) writeChunkCombined).limit(frame + 2);
```

Comment:
NIT: Would be slightly more readable as 
```suggestion
                ((Buffer) writeChunkCombined).clear();
                ((Buffer) writeChunkCombined).limit(frame + 2);
                ChunkHeader.write(writeChunkCombined, frame, remaining > transportFrameSize);
```
as well as defining `ChunkHeader.SIZE = 2` wherever it is used in this class for clarity.